src/rpg_database/character_sheet/Focus.java

```diff
@@ -0,0 +1,57 @@
+package rpg_database.character_sheet;
+
+public class Focus {
+
+	private boolean isImproved;
+	private int value;
+	private Focuses focus;
+
+	public Focus(Focuses focus) {
+		this(focus, false);
+	}
+
+	public Focus(Focuses focus, boolean isFocusImproved) {
+		if (isFocusImproved) {
+			value = 3;
+			isImproved = true;
+		} else {
+			value = 2;
+			isImproved = false;
+		}
+		this.focus = focus;
+	}
+
+	public boolean isFocusImproved() {
+		return isImproved;
+	}
+
+	public int getValue() {
+		return value;
+	}
+
+	public Focuses getFocus() {
+		return focus;
+	}
+
+	public void makeFocusImproved() {
+		setIsFocusImproved(true);
+	}
+
+	@Override
+	public boolean equals(Object obj) {

```

Comment:
Az értékeket nem hasonlítod össze??? Vagy hogy isImproved -e vagy sem? Ezt hol teszteled?

Reply:
Mert itt az volt a lényeg, hogy a hashset szempontjából megegyezző nevű focusok ne lehessenek egyenlőek, függetlenűl azok értékeitől. Tehát fel sem merülhet, hogy van két ugyan olyan focus eltérő valueval.

Reply:
Ez megint az a gondolkodás amit a cégednél csinálnak. Hadd törjem el a világodat picit: Te azzal indoklod nekem hogy írtál egy szar equals függvényt, hogy a cél az hogy fusson a Te kódod. Utána én átveszem, és teszem fel, valahol arra fogom használni, hogy két focus-t összehasonlítsak vele, mondjuk vizsgálok egy feltételt valahol, valami más adatstruktúrában, ahol össze kell 2 focust hasonlítanom. Lehet ez akár mondjuk egy specializáció is. Mi garantálja nekem hogy nem kell hosszú debuggolás után visszatúrnom a kódodba hogy miért kúródik el az enyém? Mi garantálja, hogy az equals azt csinálja amit állít a függvénynévben hogy csinál? Azt tudtad hogy az assert is ezt fogja használni előreláthatólag? Írok egy unit testet ezekre az objektumokra és assertálom őket, örülök mint majom a farkának hogy meg van hozzá írva az equals és a tesztek egészen addig fognak átverni amíg ki nem derül hogy itt van a kutya elásva.

Cserébe: ha szűrsz az értékre is, azzal elrontasz magadnak a hashset szempontjából bármit? Két különböző értékű elemet azonosnak kell venned?